packages/swapper/src/swappers/JupiterSwapper/endpoints.ts

```diff
@@ -0,0 +1,144 @@
+
+    const contractAddress =
+      step.buyAsset.assetId === solAssetId
+        ? undefined
+        : fromAssetId(step.buyAsset.assetId).assetReference
+
+    const adapter = assertGetSolanaChainAdapter(step.sellAsset.chainId)
+
+    const isManualReceiveAddress = tradeQuote.receiveAddress !== from
+
+    if (isManualReceiveAddress && step.buyAsset.assetId === solAssetId) {
+      throw Error('Manual receive address is not supported for SOL')
+    }
+
+    const { instruction: createTokenAccountInstruction, destinationTokenAccount } =
+      contractAddress && isManualReceiveAddress
+        ? await adapter.createAssociatedTokenAccountInstruction({
+            from,
+            to: tradeQuote.receiveAddress,
+            tokenId: contractAddress,
+          })
+        : { instruction: undefined, destinationTokenAccount: undefined }
+
+    const maybeSwapResponse = await getJupiterSwapInstructions({
+      apiUrl: jupiterUrl,
+      fromAddress: from,
+      toAddress: isManualReceiveAddress ? destinationTokenAccount?.toString() : undefined,
+      rawQuote: tradeQuote.rawQuote,
+      // Shared account is not supported for simple AMMs
+      useSharedAccounts:
+        tradeQuote.rawQuote.routePlan.length > 1 && isManualReceiveAddress ? true : false,
+    })
+
+    if (maybeSwapResponse.isErr()) {
+      const error = maybeSwapResponse.unwrapErr()
+      const cause = error.cause as AxiosError<any, any>
+      throw Error(cause.response!.data.detail)
+    }
+
+    const { data: swapResponse } = maybeSwapResponse.unwrap()
+
+    const convertJupiterInstruction = (instruction: Instruction): TransactionInstruction => ({
+      ...instruction,
+      keys: instruction.accounts.map(account => ({
+        ...account,
+        pubkey: new PublicKey(account.pubkey),
+      })),
+      data: Buffer.from(instruction.data, 'base64'),
+      programId: new PublicKey(instruction.programId),
+    })
+
+    const instructions: TransactionInstruction[] = [
+      ...swapResponse.setupInstructions.map(convertJupiterInstruction),
+      convertJupiterInstruction(swapResponse.swapInstruction),
+    ]
+
+    if (createTokenAccountInstruction) {
+      instructions.unshift(createTokenAccountInstruction)
+    }
+
+    if (swapResponse.cleanupInstruction) {
+      instructions.push(convertJupiterInstruction(swapResponse.cleanupInstruction))
+    }
+
+    const getFeeData = async () => {
+      const { chainNamespace } = fromAssetId(step.sellAsset.assetId)
+
+      switch (chainNamespace) {
+        case CHAIN_NAMESPACE.Solana: {
+          const sellAdapter = assertGetSolanaChainAdapter(step.sellAsset.chainId)
+          const getFeeDataInput: GetFeeDataInput<KnownChainIds.SolanaMainnet> = {
+            // Simulates a self-send
+            to: from,
+            value: '0',
+            chainSpecific: {
+              from,
+              addressLookupTableAccounts: swapResponse.addressLookupTableAddresses,
```

Comment:
I added this or we will hit the Solana TX max length, we absolutely need this to be able to send the whole TX